addons/account/models/chart_template.py

```diff
@@ -770,17 +770,23 @@ def generate_fiscal_position(self, tax_template_ref, acc_template_ref, company):
         account_template_vals = []
         for position, fp in pycompat.izip(positions, fps):
             for tax in position.tax_ids:
-                tax_template_vals.append((tax, {
-                    'tax_src_id': tax_template_ref[tax.tax_src_id.id],
-                    'tax_dest_id': tax.tax_dest_id and tax_template_ref[tax.tax_dest_id.id] or False,
-                    'position_id': fp.id,
-                }))
+                position_tax_exist = fp.tax_ids.filtered_domain([('tax_src_id', '=', tax_template_ref[tax.tax_src_id.id]),
+                                                                 ('tax_dest_id', '=', tax.tax_dest_id and tax_template_ref[tax.tax_dest_id.id] or False)])
+                if not position_tax_exist:
+                    tax_template_vals.append((tax, {
+                        'tax_src_id': tax_template_ref[tax.tax_src_id.id],
+                        'tax_dest_id': tax.tax_dest_id and tax_template_ref[tax.tax_dest_id.id] or False,
+                        'position_id': fp.id,
+                    }))
```

Comment:
This could be a lot more readable
```suggestion
                src_id = tax_template_ref[tax.tax_src_id.id]
                dest_id = tax_template_ref.get(tax.tax_dest_id.id)
                position_tax_exist = fp.tax_ids.filtered_domain([
                    ('tax_src_id', '=', src_id),
                    ('tax_dest_id', '=', dest_id),
                ])
                if not position_tax_exist:
                    tax_template_vals.append((tax, {
                        'tax_src_id': src_id,
                        'tax_dest_id': dest_id,
                        'position_id': fp.id,
                    }))
```

Reply:
improvements done. @william-andre 
ci/runbot failed because of timeout can you give a suggestion for this?

Reply:
Probably something non deterministic, I relaunched the sub build

Reply:
But I still doubt that this is the right fix, regarding my main comment of the review
https://github.com/odoo/odoo/pull/112755#pullrequestreview-1303190932

Reply:
I made a fix for the `_update_fiscal_positions_from_templates` and it's merged on https://github.com/odoo/odoo/pull/112505 and qdp-odoo suggests making pr for the account mapping as well. 
And it is right because `generate_fiscal_position` also computes while upgrading the module.